docs/feature-requests/005-blank-slate.md

```diff
@@ -0,0 +1,265 @@
+
+Clicking the "Clone GitHub repository" button opens the [Clone repository dialog](#clone-repository-dialog).
+
+### Local repository, uninitialized
+
+We detect this state when the active repository is empty, meaning the current project root has no Git repository.
+
+<img width="400" alt="github tab, local repository is uninitialized" src="https://user-images.githubusercontent.com/17565/57078458-5065a180-6cbc-11e9-94d7-f4a6f94bee54.png">
+
+#### ...no dotcom repository
+
+_Scenario:_ A user has begun a project locally and now wishes to put it under version control and share it on GitHub.
+
+Clicking the "Create GitHub repository" button opens the [Create repository dialog](#create-repository-dialog) in "local repository" mode.
+
+### Local repository, initialized, no dotcom remotes
+
+We detect this state when the active repository is present but has no dotcom remotes.
+
+<img width="400" alt="github tab, local repository with no GitHub remotes" src="https://user-images.githubusercontent.com/17565/57078536-72f7ba80-6cbc-11e9-84e8-8c2384caccdc.png">
+
+
+#### ...no dotcom repository
+
+_Scenario:_ A user has begun a project locally and now wishes to share it on GitHub.
+
+Clicking the "Create GitHub repository" button opens the [Create repository dialog](#create-repository-dialog) panel in "local repository" mode.
+
+#### ...existing dotcom repository
+
+_Scenario:_ A user has a project cloned locally and it exists on GitHub.
+
+_Scenario:_ A user has a project cloned locally and it exists on GitHub, but they cannot push. Maybe they cloned the project before to read its source more easily, but now they wish to contribute.
+
+Clicking the "Connect GitHub repository" button opens the [Connect repository dialog](#connect-repository-dialog).
+
+### Local repository, initialized, dotcom remotes
+
+This is the state we handle now: when an active repository is present and has one or more dotcom remotes.
+
+## Clone repository dialog
+
+The clone repository dialog begins in search mode. As you type within the text input, once more than three characters have been entered, repositories on GitHub matching the entered text appear in the result list below. Repositories may be identified by full clone URL, `owner/name` pair, or a unique substring of `owner/name`.
+
+<img width="650" alt="clone dialog, empty search" src="https://user-images.githubusercontent.com/17565/57096303-f5937080-6ce2-11e9-8d71-e63677dcba7c.png">
+
+<img width="650" alt="clone dialog, search results" src="https://user-images.githubusercontent.com/17565/57096339-07751380-6ce3-11e9-92ab-4714d6c9df8a.png">
+
+### GitHub clone mode
+
+Clicking on an entry in the search result list or entering the full clone URL of a GitHub repository changes the dialog to "GitHub clone" mode:
+
+<img width="650" alt="clone dialog, GitHub mode" src="https://user-images.githubusercontent.com/17565/57244555-ac0a9480-7006-11e9-8048-950e58f038a3.png">
```

Comment:
We can include a button here to let users choose on a GUI where to clone the repo into.

Reply:
Oh like a `...` button that opens a folder chooser? Good call 👍 

One sort-of complication is that we need to prompt for a _new_ folder and I don't think our OS integration lets us specify that. Might be okay to detect the "existing non-empty folder" case after the picker is dismissed and show an "invalid form control" state?

Reply:
Done. Also relaxed the destination path validity description to accept an existing but empty directory, which you can clone into.